Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -183,6 +269,28 @@ private Uri GetAppIdEndPointUri(string instrumentationKey)
             return new Uri(this.endpointAddress, string.Format(CultureInfo.InvariantCulture, AppIdQueryApiRelativeUriFormat, instrumentationKey));
         }
 
+        /// <summary>
+        /// Registers failure for further action in future.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which the failure occurred.</param>
+        /// <param name="ex">Exception indicating failure.</param>
+        private void RegisterFailure(string instrumentationKey, Exception ex)
```

Comment:
we can move logic for AggregateException to this method and it'll make code simpler